File: 11/taehyuk/1334.py
class Solution:
    def findTheCity(self, n: int, edges: List[List[int]], distanceThreshold: int) -> int:
        graph = [[float('inf')] * n for _ in range(n)]
        
        for edge in edges:
            u, v, w = edge
            graph[u][v] = w
            graph[v][u] = w
        
        for i in range(n):
            graph[i][i] = 0
        
        for k in range(n):
            for i in range(n):
                for j in range(n):
                    if graph[i][j] > graph[i][k] + graph[k][j]:
                        graph[i][j] = graph[i][k] + graph[k][j]
        
        minReachable = n
        maxCity = -1
        
        for i in range(n):
            count = 0
            for j in range(n):
                if graph[i][j] <= distanceThreshold:
                    count += 1
            if count <= minReachable:
                minReachable = count
                maxCity = i
        
        return maxCity

File: 11/ahnjaewoo/1334.cpp
class Solution {
public:
    int findTheCity(int n, vector<vector<int>>& edges, int distanceThreshold) {
        int inf = 100000000;
        vector<vector<int>> distance(n, vector<int>(n, inf));
        for (int i = 0; i < edges.size(); i++) {
            distance[edges[i][0]][edges[i][1]] = edges[i][2];
            distance[edges[i][1]][edges[i][0]] = edges[i][2];
        }
        for (int i = 0; i < n; i++) {
            distance[i][i] = 0;
        }
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    if (distance[j][k] > distance[j][i] + distance[i][k]) {
                        distance[j][k] = distance[j][i] + distance[i][k];
                    }
                }
            }
        }
        int min = n + 1;
        int minNode = -1;
        for (int i = 0; i < n; i++) {
            int current = 0;
            for (int j = 0; j < n; j++) {
                if (i != j && distance[i][j] <= distanceThreshold && distance[i][j] != 0) {
                    current += 1;
                }
            }
            if (current <= min) {
                minNode = i;
                min = current;
            }
        }

        return minNode;
    }
};